JSON.md

```diff
@@ -113,11 +113,86 @@ The 3 build steps above achieve 3 distinct things:
+All fields are required unless otherwise noted.
+Please use `cfbs validate` while editing `cfbs.json` files manually - we won't attempt to list absolutely all the validation rules here.
+
+### Top level keys
+
+At the top level of a `cfbs.json` file, these fields are available:
+
+* `name` (string): The human readable name of the project.
+  An example could be: `Northern.tech production policy`
+* `description` (string): Human readable description of what this project is for.
+  For example: `This project builds the policy set we deploy to all production hosts.`
+* `type` (string): What kind of project this is.
+  One of: `policy-set` (default), `index`, and `module`.
+  If you are setting up a project to build a policy set (to deploy on a hub), use `policy-set`.
+  For developing a new module (or multiple) to publish on [build.cfengine.com](https://build.cfengine.com), or to use in your other projects, use `module`.
+  To set up an alternate list of modules (instead of relying on [the default one on GitHub](https://github.com/cfengine/build-index/blob/master/cfbs.json)), use `index`.
+* `index` (string or dictionary): URL, relative path, or inline dictionary.
+  Used by `cfbs add` and `cfbs search`, to know where to look for modules.
+  Required and must be dictionary if the `type` is `index`, optional otherwise.
+  When it's a dictionary, the keys must be the unique module name which will be converted to the module's `name` field when added to the `build` array.
+* `git` (true or false): Whether `cfbs` should make git commits after editing `./cfbs.json` and related files.
+  Optional, defaults to false.
+* `provides` (dictionary of modules): Which modules this repo provides when someone tries to add it by URL (`cfbs add https://github.com/some/repo`).
+  Required for `cfbs add <URL>` to work, optional otherwise.
+  Most commonly used for projects with type `module`.
+  The keys must be the unique module name which will be converted to the module's `name` field when added to the `build` array.
+* `build` (list of modules): The modules to combine into a policy set when running `cfbs build`.
+  Required and must be non-empty for `policy-set` type and also for `cfbs build` command to work, optional otherwise.
+  (Even if you are developing a `module`, it is useful to be able to put modules in `build`, to build and deploy a policy set to test).
+
+### Module level keys
+
+The modules inside `build`, `provides`, and `index` use these fields:
+
+* `alias` (string): Used to rename a module in an index, or to provide a short name alternative.
+  Gets translated to the value (real module name) by `cfbs add`.
+  Only valid inside `index`.
+  Optional, must be the only field if used.
+* `name` (string): The unique name of the module (unique within project and within index).
+  For `provides` and `index` dictionaries, this name must be the key of each entry (not a field inside).
```

Comment:
Took me a long time to understand this sentence. But I'm not sure how to make it more clear.